web/src/features/course/components/course-editor-sections.tsx

```diff
@@ -1,64 +1,247 @@
-import { forwardRef, useImperativeHandle } from "react";
+import { Plus, ChevronLeft, ChevronDown, Trash2, Menu } from "lucide-react";
+import { useState, useEffect } from "react";
 import { useTranslation } from "react-i18next";
 
 import { Button } from "@/shared/components/shadcn/button";
+import { Card, CardContent} from "@/shared/components/shadcn/card";
+import { 
+  CourseSectionCreateFunction,
+  CourseSectionDeleteFunction, 
+  CourseSectionQueryFunction, 
+  CourseSectionSetPublish, 
+  CourseSectionUpdateFunction} from "@/course/api/course-sections-api";
+import { useQuery } from "@tanstack/react-query";
+import { CourseSection } from "@/shared/api/types.gen";
+import GlobalLoader from "@/shared/components/global-loader";
+import { SectionForm } from "./SectionForm";
 
 interface CourseEditorSectionsProps {
-  courseId?: string;
+  courseId: string;
   onComplete?: () => void;
+  onGoBack?: () => void;
 }
 
-export interface CourseEditorSectionsRef {
-  isDirty: () => boolean;
-}
-
-const CourseEditorSections = forwardRef<
-  CourseEditorSectionsRef,
-  CourseEditorSectionsProps
->(({ courseId, onComplete }, ref) => {
+const CourseEditorSections = ({ courseId, onComplete, onGoBack }: CourseEditorSectionsProps) => {
   const { t } = useTranslation();
+  const [currentSectionEditing, setCurrentSectionEditing] = useState<string | null>(null);
+  const [isCreating, setIsCreating] = useState(false);
+  
+  const createMutation = CourseSectionCreateFunction();
+  const updateMutation = CourseSectionUpdateFunction();
+  const deleteMutation = CourseSectionDeleteFunction();
+  const setPublishMutation = CourseSectionSetPublish();
+  
+  const {
+    data: queryCourseSections,
+    isLoading: queryIsLoading,
+    refetch,
+  } = useQuery({
+    ...CourseSectionQueryFunction(courseId ?? ""),
+    enabled: true,
+  });
+  const [sections, setSections] = useState<Array<CourseSection> | undefined>();
+
+  // Update course sections
+  useEffect(() => {
+    const run = () => {
+      refetch().then(() => {
+        setSections([...(queryCourseSections) ?? []]);
+      });
+    };
+    run();
+  }, [currentSectionEditing, isCreating, queryIsLoading]);
```

Comment:
there are some missing dependencies here. Needs to be:
```suggestion
}, [currentSectionEditing, isCreating, queryIsLoading, refetch, queryCourseSections]);
```

Reply:
I thought this would cause the `useEffect()` to run infinitely, but it does not, so i have changed it the way you suggested